langs/java.go

```diff
@@ -184,50 +190,121 @@ func (h *JavaLangHelper) getFDKAPIVersion() (string, error) {
 		return h.latestFdkVersion, nil
 	}
 
-	const versionURL = "https://api.bintray.com/search/packages/maven?repo=fnproject&g=com.fnproject.fn&a=fdk"
-	const versionEnv = "FN_JAVA_FDK_VERSION"
-	fetchError := fmt.Errorf("Failed to fetch latest Java FDK javaVersion from %v. Check your network settings or manually override the javaVersion by setting %s", versionURL, versionEnv)
+	const bintrayVersionURL = "https://api.bintray.com/search/packages/maven?repo=fnproject&g=com.fnproject.fn&a=fdk"
+	const mavenVersionUrl = "https://repo1.maven.org/maven2/com/fnproject/fn/fdk/maven-metadata.xml"
 
-	type parsedResponse struct {
-		Version string `json:"latest_version"`
-	}
+	const versionEnv = "FN_JAVA_FDK_VERSION"
+	fetchError := fmt.Errorf("Failed to fetch latest Java FDK javaVersion. Check your network settings or manually override the javaVersion by setting %s", versionEnv)
 	version := os.Getenv(versionEnv)
+
 	if version != "" {
 		return version, nil
 	}
+	version, err := h.getFDKLastestFromURL(mavenVersionUrl, bintrayVersionURL)
+	if err != nil {
+		return "", fetchError
+	}
 
-	// nishalad95: bin tray TLS certs cause verification issues on OSX, skip TLS verification
+	h.latestFdkVersion = version
+	return version, nil
+}
+
+func (h *JavaLangHelper) getFDKLastestFromURL(comURL string, bintrayURL string) (string, error) {
+	var buf *bytes.Buffer
+	var err error
+	err = fmt.Errorf("All URL failed to respond ")
+
+	//First search for com.fnproject.fn from Maven Central to get the latest version
+	buf, err = h.getURLResponse(comURL, false)
+	if err == nil {
+		version, e1 := h.parseMavenResponse(*buf)
+		if e1 == nil {
+			h.pomType = "maven"
+			return version, e1
+		}
+	}
+
+	//Second time search for com.fnproject.fn from Bintray to get the latest version, if fetch from Maven fails
+	buf, err = h.getURLResponse(bintrayURL, true)
+	if err == nil {
+		version, e1 := h.parseBintrayResponse(*buf)
+		if e1 == nil {
+			h.pomType = "bintray"
+			return version, e1
+		}
+	}
+
+	//In all other case return error as latest FDK version is not identified
+	return "", err
+}
```

Comment:
Consider accepting variadic params of structs containing type & url and loop over it? It will prevent code duplication.

Reply:
Since we are never likely to have dynamic number of package repos, this change is good.